lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -121,44 +80,71 @@ def non_persistent_schedules
       #                                     :every => '15mins',
       #                                     :queue => 'high',
       #                                     :args => '/tmp/poop'})
-      def set_schedule(name, config)
+      #
+      # Preventing a reload is optional and available to batch operations
+      def set_schedule(name, config, reload = true)
         persist = config.delete(:persist) || config.delete('persist')
-        redis.pipelined do
-          redis.hset(:schedules, name, encode(config))
-          redis.sadd(:schedules_changed, name)
-          redis.sadd(:persisted_schedules, name) if persist
+
+        if persist
+          redis.hset(keys[:persistent_schedules], name, encode(config))
+        else
+          non_persistent_schedules[name] = decode(encode(config))
         end
-        config
+
+        redis.sadd(keys[:schedules_changed], name)
+        reload_schedule! if reload
       end
 
       # retrive the schedule configuration for the given name
       def fetch_schedule(name)
-        decode(redis.hget(:schedules, name))
-      end
-
-      def schedule_persisted?(name)
-        redis.sismember(:persisted_schedules, name)
+        schedule[name]
       end
 
       # remove a given schedule by name
       def remove_schedule(name)
-        redis.hdel(:schedules, name)
-        redis.srem(:persisted_schedules, name)
-        redis.sadd(:schedules_changed, name)
+        non_persistent_schedules.delete(name)
+        redis.hdel(keys[:persistent_schedules], name)
+        redis.sadd(keys[:schedules_changed], name)
+
+        reload_schedule!
+      end
+
+      def keys
+        lock_prefix = ENV['RESQUE_SCHEDULER_MASTER_LOCK_PREFIX'] || ''
```

Comment:
Maybe this change should be handled in a separate PR.  I'm not really clear on why just the lock would be namespaced, rather than all the scheduler's data or all of the resque redis data.  Either way, it seems like this ENV var isn't appropriately named after a change like this, and should probably be read with the other ENV vars in Resque::Scheduler::Configuration